packages/signers/ledger-signer/src/get-metadata.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Option,
+  Bytes,
+  metadata,
+  compact,
+  Tuple,
+  V15,
+} from "@polkadot-api/substrate-bindings"
+import { HexString } from "@polkadot-api/substrate-bindings"
+
+const opaqueBytes = Bytes()
+const optionOpaque = Option(opaqueBytes)
+const opaqueOpaqueBytes = Tuple(compact, opaqueBytes)
+
+const getAnyMetadata = (input: Uint8Array | HexString) => {
+  try {
+    return metadata.dec(input)
+  } catch (_) {}
+
+  // comes from metadata.metadata_at_version
+  try {
+    return metadata.dec(optionOpaque.dec(input)!)
+  } catch (_) {}
+
+  // comes from state.getMetadata
+  try {
+    return metadata.dec(opaqueBytes.dec(input))
+  } catch (_) {}
+
+  // comes from metadata.metadata
+  try {
+    return metadata.dec(opaqueOpaqueBytes.dec(input)[1])
+  } catch (_) {}
+
+  throw null
+}
```

Comment:
Not really. We can go for regular metadata codec. Actually, I'd say that it might be interesting for all signers to accept any metadata format. This would simplify IMO passing the metadata from outside to a signer.